### DISCUSSION TITLE
Not getting entity

### DISCUSSION THREAD

**Discussion by wuuthraad-x** [explosion/spaCy#11109] [Help: Model Advice] [answered]
I am using google colab
I tried to get the entity of a specific word and it did not work on the en_core_web_sm(It did not return an entity it was just blank). I though that the issue was the model itself(not being trained on that much data). I later downloaded the en_core_web_lg model and it gave me the exact same issue  am I doing something wrong? It works on most words but confuses me why it does not on certain words

!python -m spacy download en_core_web_lg

smol_nlp = spacy.load('en_core_web_sm')
text = 'KAPS PARKING'
print(smol_nlp(text).ents)

nlp = spacy.load('en_core_web_lg')
text = 'KAPS PARKING'
print(nlp(text).ents)

**Comment by wuuthraad-x** [comment]
I also tried

text = 'KAPS PARKING'
for words in nlp(text).ents:
  print(words , words.label_)

**Reply by polm** [reply]
The pretrained pipelines are trained on data like newspaper articles, which have complete sentences and proper casing. They rely not only on the words in the entity but on context (surrounding words). So short, all-caps text is not something they've seen before and they won't work very well. If your data is mostly like that you may need to train your own model to get satisfying results. 

Additionally, as a human I'm not sure what entity you would even expect for this. Is KAPS PARKING an ORG? Maybe a LOC? Or maybe KAPS is a PERSON? If your task is not clear to a human it can be hard to teach a machine.

**Reply by wuuthraad-x** [reply]
OK , I'll try to put all the words in lowercase and maybe even manually add labels. Thanks for the information I did not know that.

**Comment by wuuthraad-x** [comment, marked as answer]
changing the values to lowercase helped!. It is labelling the value(which is now "kaps parking" instead of "KAPS PARKING") and an ORG(which it is)

I first changed all values in the column to lowercase
then tried to get the entity value

train[columns_name] = train[column_name].apply(str.lower)

for words in nlp(train[column_name][0]).ents:
  print(words, words.label_)
